mypyc/test-data/irbuild-classes.test

```diff
@@ -1370,3 +1370,14 @@ class NonNativeClassContradiction():  # E: Class is marked as native_class=True
 @mypyc_attr(native_class="yes")
 class BadUse():  # E: native_class must be used with True or False only
     pass
+
+[case testMypycAttrNativeClassMetaError]
+from mypy_extensions import mypyc_attr
+
+@mypyc_attr(native_class=True)
+class M(type):  # E: Inheriting from most builtin types is unimplemented
+    pass
+
+@mypyc_attr(native_class=True)
+class A(metaclass=M):  # E: Class is marked as native_class=True but it can't be a native class
```

Comment:
Maybe also mention in the error message that the metaclass is not supported with a native class?

Reply:
Let me separate this out to its own PR. Right now `is_implicit_extension_class` (in mypyc/irbuild/util.py) just returns True/False. To print a nice message we would have to go through each individual check in `is_implicit_extension_class` and make a nice user-facing message.
It is very good feedback, and I agree with you, just think it can be split out to separate PR.